### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by rahulbcn27**
Migration from Travis CI to Github Actions CI . I created a new yml file in .github/workflows doing the same task that travis.yml 
use to do.

### CODE REVIEW

File: .github/workflows/github-actions.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Github-Actions
```
Comment (rohityadavcloud) [inline .github/workflows/github-actions.yml:1]: Missing license header @rahulbcn27

File: .github/workflows/github-actions.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+on:
+  push:
+    branches: [ "github-actions" ]
```
Comment (rohityadavcloud) [inline .github/workflows/github-actions.yml:5]: Should this be only branch push?

File: .github/workflows/github-actions.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.14.x
```
Comment (rohityadavcloud) [inline .github/workflows/github-actions.yml:18]: Go version seems outdated?

File: .github/workflows/github-actions.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    - name: Run Script
+      run:  make all
```
Comment (rohityadavcloud) [inline .github/workflows/github-actions.yml:21]: Do we assume `make` is available or need to install any dependency?

**Review by rohityadavcloud — CHANGES_REQUESTED** [review]
Changes requested

File: .github/workflows/github-actions.yml (hunk 4 above)
Comment (rahulbcn27) [reply .github/workflows/github-actions.yml:21]: Here I assume make is available.

File: .github/workflows/github-actions.yml (hunk 3 above)
Comment (rahulbcn27) [reply .github/workflows/github-actions.yml:18]: Updated to 1.19.x

File: .github/workflows/github-actions.yml (hunk 2 above)
Comment (rahulbcn27) [reply .github/workflows/github-actions.yml:5]: Fixed it

File: .github/workflows/github-actions.yml (hunk 1 above)
Comment (rahulbcn27) [reply .github/workflows/github-actions.yml:1]: Added.

File: .github/workflows/github-actions.yml [hunk 5]
```diff
@@ -15,13 +15,23 @@
...
 
-sudo: false
-language: go
-go:
-  - 1.14.x
+name: Github-Actions
 
-script:
-  - make all
+on: [push]
```
Comment (rohityadavcloud) [inline .github/workflows/github-actions.yml:20]: Can you check Github Actions workflow for apache/cloudstack. We want (a) this to run on pull requests and (b) have concurrency limit (this ensure when too many commits are push on the branch or PR, it cancels older builds).

**Review by rohityadavcloud — CHANGES_REQUESTED** [review]
Better, just one outstanding issue.

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
It appears we can't test your config change without merging it. Let's do that.

File: .github/workflows/github-actions.yml [hunk 6]
```diff
@@ -15,13 +15,27 @@
...
 
-env:
-  - GO111MODULE=on
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (rohityadavcloud) [inline .github/workflows/github-actions.yml:28]: @rahulbcn27 LGTM, Github Actions seems to be working. Can you raise a new PR to change runs-on to ubuntu-22.04 (see apache/cloudstack for example).

Comment (rahulbcn27) [reply .github/workflows/github-actions.yml:28]: https://github.com/rahulbcn27/cloudstack-cloudmonkey/pull/1

**Pull request merged**